nemo/core/optim/lr_scheduler.py

```diff
@@ -412,6 +412,9 @@ def prepare_lr_scheduler(
         scheduler_config = OmegaConf.to_container(scheduler_config, resolve=True)
 
     # Test to see if config follows above schema
+
+    addMaxArgsFlag = True
+    interval = 'step'
```

Comment:
Can you describe the changes in this file?

Reply:
These are general changes to enable reduceLR OnPlateau scheduler.
the max_steps argument does not fit into this scheduler initialization so it should not be added.
in addition, it does not work with per-step evaluation of learning rate, it is too noisy.

An alternative is to abandon reduceLrOn Plateau scheduler and use another scheduler that gives about the same learning rate (as we know from the learning logger I have mistakenly commited)

Reply:
@titu1994, can you ask you to review just this file?

Reply:
Ah just realised I was supposed to review just this file. Oh well. This is a reasonable change and we do want to support external schedulers so it's good to add this support.